### PR TITLE
Indicate spent resource in trade log

### PR DESCRIPTION
**Pull request by nwai90**
Closes https://github.com/bafolts/terraforming-mars/issues/2965

### CODE REVIEW

File: src/Player.ts [hunk 1]
```diff
@@ -1413,27 +1413,30 @@ export class Player implements ISerializable<SerializedPlayer> {
...
         if (colony.name === colonyName) {
-          this.game.log('${0} traded with ${1}', (b) => b.player(this).colony(colony));
           if (payWith === Resources.MEGACREDITS) {
             this.game.defer(new SelectHowToPayDeferred(
               this,
               mcTradeAmount,
               {
                 title: 'Select how to pay ' + mcTradeAmount + ' for colony trade',
                 afterPay: () => {
+                  this.game.log('${0} spent ${1} MC to trade with ${2}', (b) => b.player(this).number(mcTradeAmount).colony(colony));
```
Comment (vsrisuknimit) [inline src/Player.ts:1423]: Should we fix the translation files as well for all these different new log messages?

**Comment by nwai90** [comment]
I will add placeholders for them. @alrusdi can probably help with the correct Russian translation. Once that is done I think it's fine to merge as is and add other localisations later. None of the other languages actually have complete translations at this point.

**Review by vsrisuknimit — APPROVED** [review]
lgtm

File: src/Player.ts [hunk 2]
```diff
@@ -1413,27 +1413,30 @@ export class Player implements ISerializable<SerializedPlayer> {
...
                   colony.trade(this);
                 },
               },
             ));
           } else if (payWith === Resources.ENERGY) {
             this.energy -= energyTradeAmount;
+            this.game.log('${0} spent ${1} energy to trade with ${2}', (b) => b.player(this).number(energyTradeAmount).colony(colony));
```
Comment (kberg) [inline src/Player.ts:1430]: If you feel like reducing the repeated string, I'd be grateful.

**Pull request merged**